apis/core.oam.dev/v1alpha2/core_types.go

```diff
@@ -412,6 +412,16 @@ type WorkloadTrait struct {
 
 	// Message will allow controller to leave some additional information for this trait
 	Message string `json:"message,omitempty"`
+
+	// AppliedGeneration indicates the generation observed by the appConfig controller.
+	// The same field is also recorded in the annotations of traits.
+	// A trait is possible to be deleted from cluster after created.
+	// This field is useful to track the observed generation of traits after they are
+	// deleted.
+	AppliedGeneration int64 `json:"appliedGeneration,omitempty"`
+
+	// DependencyUnsatisfied notify does the trait has dependency unsatisfied
+	DependencyUnsatisfied bool `json:"dependencyUnsatisfied,omitempty"`
```

Comment:
normally it's better for a bool name to be positive (DependencySatisfied) so we don't need to use double negative

Reply:
it will be uncompatiable, the default case should be false which is satisfied